spec/repositories/manual_repository_spec.rb

```diff
@@ -139,10 +139,19 @@
   end
 
   describe "#[]" do
+    let(:section_repository) { double(:section_repository) }
+
     before do
       allow(record_collection).to receive(:find_by).and_return(manual_record)
       allow(manual_record).to receive(:latest_edition).and_return(edition)
       allow(Manual).to receive(:new).and_return(manual)
+      allow(SectionRepository).to receive(:new).with(manual: manual).and_return(section_repository)
+      allow(manual).to receive(:'sections=')
+      allow(manual).to receive(:'removed_sections=')
+      allow(edition).to receive(:section_ids).and_return([:section_id])
+      allow(edition).to receive(:removed_section_ids).and_return([:removed_section_id])
+      allow(section_repository).to receive(:fetch).with(:section_id).and_return(:section)
+      allow(section_repository).to receive(:fetch).with(:removed_section_id).and_return(:removed_section)
```

Comment:
Minor: Having this number of stubs is a bit of an indicator that the code under test may not be in the best shape. However, I imagine this is what you mean when you wrote the following in the PR description:

> I've had to re-write the unit tests that were on the SectionAssociationMarshaller to fit with the setup of the ManualRepository specs. This has left the spec file in a bit of a mess, but I wanted to try to make it as clear as possible the provenance of these specs. I've decided to defer refactoring / tidying up ManualRepository for now, I think that can be a separate PR.

One way I can imagine making this simpler is to use an actual `Manual` instance with real sections, rather than stubbing everything. However, I'm happy that this can be done separately later.